### DISCUSSION TITLE
Deployment options for .Net 6 ExcelDna AddIn

### DISCUSSION THREAD

**Discussion by JonTheTurnip** [Excel-DNA/ExcelDna#607] [Q&A] [answered]
Hi there,

I tried running my .Net 6 ExcelDna Addin on a machine without .net 6 installed and unsurprisingly I get the error message below.

Is there a way to publish such that the xll is self contained, and does not have any .net runtime prerequisites?

![image](https://github.com/Excel-DNA/ExcelDna/assets/10959562/8d49977a-ac90-419c-b864-51133d8c28d2)

**Comment by govert** [comment, marked as answer]
We don't support the bring-your-own-runtime option of .NET Core, because the add-in isolation mechanism we use (the AssemblyLoadContext etc.) doesn't support that. Because only one version of a .NET Core runtime can be loaded into the Excel process (in addition to the .NET Framework runtime) separate add-ins can't bring their own version of the .NET 6 runtime. So you need a .NET 6 runtime installed on the machine before your add-in can load. This installation does not need admin permissions to do and can be in a directory related to your add-in.

You could make an installer for your add-in that adds the .NET 6 runtime as a prerequisite, for example based on the WixInstaller template here https://github.com/Excel-DNA/WiXInstaller or the updated test version here https://github.com/govert/Test-WiXInstaller. I doesn't have the .NET 6 dependency at the moment, but I guess that would be a standard thing to add to a WiX project.

In a future version of .NET we might try to support the NativeAOT compilation more of .NET, and then you would not need a runtime installation at all. But there will be limitations on the supported features. This works is still a while away though, and will depend on whether we continue to get generous users signing up as [GitHub Sponsors](https://github.com/sponsors/Excel-DNA/) to support further development.

**Reply by JonTheTurnip** [reply]
Thank you for the detailed response, much appreciated.  

I'll have a look into the other options you mention, they both sound workable.
